lib/ansible/modules/cloud/vmware/vmware_portgroup.py

```diff
@@ -196,297 +243,785 @@
         hosts = self.params['hosts']
         cluster = self.params['cluster_name']
-        self.portgroup_name = self.params['portgroup_name']
-        self.switch_name = self.params['switch_name']
+        self.portgroup = self.params['portgroup']
+        self.switch = self.params['switch']
         self.vlan_id = self.params['vlan_id']
-        self.promiscuous_mode = self.params['network_policy'].get('promiscuous_mode')
-        self.forged_transmits = self.params['network_policy'].get('forged_transmits')
-        self.mac_changes = self.params['network_policy'].get('mac_changes')
-        self.network_policy = self.create_network_policy()
+        if self.params['security']:
+            self.sec_promiscuous_mode = self.params['security'].get('promiscuous_mode')
+            self.sec_forged_transmits = self.params['security'].get('forged_transmits')
+            self.sec_mac_changes = self.params['security'].get('mac_changes')
+        else:
+            self.sec_promiscuous_mode = None
+            self.sec_forged_transmits = None
+            self.sec_mac_changes = None
+        if self.params['traffic_shaping']:
+            self.ts_enabled = self.params['traffic_shaping'].get('enabled')
+            self.ts_average_bandwidth = self.params['traffic_shaping'].get('average_bandwidth')
+            self.ts_peak_bandwidth = self.params['traffic_shaping'].get('peak_bandwidth')
+            self.ts_burst_size = self.params['traffic_shaping'].get('burst_size')
+        else:
+            self.ts_enabled = None
+            self.ts_average_bandwidth = None
+            self.ts_peak_bandwidth = None
+            self.ts_burst_size = None
+        if self.params['teaming']:
+            self.teaming_load_balancing = self.params['teaming'].get('load_balancing')
+            self.teaming_failure_detection = self.params['teaming'].get('network_failure_detection')
+            self.teaming_notify_switches = self.params['teaming'].get('notify_switches')
+            self.teaming_failback = self.params['teaming'].get('failback')
+            self.teaming_failover_order_active = self.params['teaming'].get('active_adapters')
+            self.teaming_failover_order_standby = self.params['teaming'].get('standby_adapters')
+            if self.teaming_failover_order_active is None:
+                self.teaming_failover_order_active = []
+            if self.teaming_failover_order_standby is None:
+                self.teaming_failover_order_standby = []
+        else:
+            self.teaming_load_balancing = None
+            self.teaming_failure_detection = None
+            self.teaming_notify_switches = None
+            self.teaming_failback = None
+            self.teaming_failover_order_active = None
+            self.teaming_failover_order_standby = None
+        # NOTE: the following options are deprecated and should be removed in 2.11
+        self.teaming_inbound_policy = self.module.params['teaming']['inbound_policy']
```

Comment:
Through reintroduction of `inbound_policy` and `rolling_order`, 295 and 296 breaks the module.  These lines are assuming a teaming config is defined.  I get the following error:

```
"msg": "'NoneType' object has no attribute '__getitem__'"

The full traceback is:
  File "/tmp/ansible_rDL_Y6/ansible_module_vmware_portgroup.py", line 1023, in main
    host_portgroup = VMwareHostPortGroup(module)
  File "/tmp/ansible_rDL_Y6/ansible_module_vmware_portgroup.py", line 295, in __init__
    self.teaming_inbound_policy = self.module.params['teaming']['inbound_policy']
```

Reply:
I need to keep those options for compatibility. Otherwise, the PR won't be accepted. Unfortunately, they cannot be removed easily. I changed the code. Now it should work. Can you try again?

Reply:
Looks good, passes my tests. 👍

Reply:
thx